test/dtest/ccmlib/scylla_cluster.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from __future__ import annotations
+
+from test.pylib.manager_client import ManagerClient
+from test.dtest.ccmlib.scylla_node import ScyllaNode
+
+
+class ScyllaCluster:
+    scylla_mode = "debug"  # TODO: implement this
+
+    def __init__(self, manager: ManagerClient):
+        self.manager = manager
+        self._config_options = {}
+
+    @property
+    def nodes(self) -> dict[str, ScyllaNode]:
+        return {node.name: node for node in self.nodelist()}
+
+    def nodelist(self) -> list[ScyllaNode]:
+        return [ScyllaNode(cluster=self, server=server) for server in self.manager.all_servers()]
+
+    def populate(self, nodes: int) -> ScyllaCluster:
+        self.manager.servers_add(servers_num=nodes, config=self._config_options, start=False)
+        return self
+
+    def start(self, wait_for_binary_proto: bool | None = None, wait_other_notice: bool | None = None) -> None:
+        for server in self.manager.all_servers():
+            self.manager.server_start(server_id=server.server_id)
```

Comment:
I'm confused what I'm seeing here. I thought that the ScyllaManager stuff must use async and "await". How do you not need to do all these ugly awaits here? How did you avoid your own functions becoming "infected" with async?

Reply:
Oh, I see the universialasync somehow does this. I have no idea how :-)